### PR TITLE
Center Grid around Viewbox on Viewbox Changed

### LINKED ISSUE #339: Center Grid around Viewbox on Viewbox Changed
Currently, the grid is always centered around 0, 0 and never updated. If a user scrolls the canvas too far he will see no grid anymore. To prevent that the grid should always be centered around the viewbox after the viewbox changed.

### PR DESCRIPTION
**Pull request by philippfromme**
Closes #339

### CODE REVIEW

**Comment by philippfromme** [comment]
Codecoverage doesn't like me. 😢

File: lib/features/grid-snapping/Grid.js [hunk 1]
```diff
@@ -80,6 +93,23 @@ Grid.prototype._init = function() {
   });
 };
 
+Grid.prototype._centerGridAroundViewbox = function(viewbox) {
+  if (!viewbox) {
+    viewbox = this._canvas.viewbox();
+  }
+
+  if (!viewbox) {
+    return;
```
Comment (barmac) [inline lib/features/grid-snapping/Grid.js]: Can this ever happen? `Canvas#viewbox` always returns something so I think this check is unnecessary.

**Comment by nikku** [comment]
I suggest to not merge this PR if we go for https://github.com/bpmn-io/diagram-js/pull/361.

**Comment by philippfromme** [comment]
I'd suggest to still add this improvement to the separate grid library.

**Comment by nikku** [comment]
This PR would need to get updated now as https://github.com/bpmn-io/diagram-js/pull/361 is merged. I could do that, however I got to focus on a few other issues before.

Alternative (sketched above): Close this PR :walking_man:

**Comment by philippfromme** [comment]
@nikku I'll update the PR. 💪

File: lib/features/grid-snapping/Grid.js (hunk 1 above)
Comment (philippfromme) [reply lib/features/grid-snapping/Grid.js]: 👍

**Comment by philippfromme** [comment]
@barmac Ready 2 be reviewed.

**Pull request merged**